### PR TITLE
3.6 Node updateTransform apply recursive form

### PR DESCRIPTION
**Pull request by PatriceJiang**
Re-commit:  #12017

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by PatriceJiang** [comment]
@cocos-robot run test cases

File: native/cocos/core/scene-graph/Node.cpp [hunk 1]
```diff
@@ -521,70 +499,60 @@ void Node::setScaleInternal(float x, float y, float z, bool calledFromJS) {
...
             }
-        } else if (child) {
-            if (dirtyBits & static_cast<uint32_t>(TransformBit::POSITION)) {
-                currChild->_worldPosition.set(currChild->_localPosition);
-                currChild->_worldMatrix.m[12] = currChild->_worldPosition.x;
-                currChild->_worldMatrix.m[13] = currChild->_worldPosition.y;
-                currChild->_worldMatrix.m[14] = currChild->_worldPosition.z;
+            Quaternion quat = _worldRotation;
+            quat.conjugate();
+            Mat3 m3Out(quat);
```
Comment (SantyWang) [inline native/cocos/core/scene-graph/Node.cpp:533]: use static instead?

File: native/cocos/core/scene-graph/Node.cpp [hunk 2]
```diff
@@ -606,27 +574,17 @@ Mat4 Node::getWorldRT() {
...
-        if (cur == nullptr) {
-            continue;
+    const uint32_t hasChangedFlags = getChangedFlags();
+    const uint32_t dirtyFlags = getDirtyFlag();
+    if (isValid() && (dirtyFlags & hasChangedFlags & curDirtyBit) != curDirtyBit) {
+        setDirtyFlag(dirtyFlags | curDirtyBit);
+        setChangedFlags(hasChangedFlags | curDirtyBit);
+
+        for (Node *child : getChildren()) {
+            child->invalidateChildren(dirtyBit | TransformBit::POSITION);
```
Comment (SantyWang) [inline native/cocos/core/scene-graph/Node.cpp:586]: why add TransformBit::POSITION

File: native/cocos/core/scene-graph/Node.cpp (hunk 1 above)
Comment (PatriceJiang) [reply native/cocos/core/scene-graph/Node.cpp:533]: static variables may cause more cache missing

File: native/cocos/core/scene-graph/Node.cpp (hunk 2 above)
Comment (PatriceJiang) [reply native/cocos/core/scene-graph/Node.cpp:586]: The original procedure did this too. 

If a node updates its scale or rotation, all children's position is also affected.

**Review by SantyWang — APPROVED** [review]

**Comment by minggo** [comment]
@cocos-robot run test cases

**Review by minggo — APPROVED** [review]

**Comment by minggo** [comment]
@cocos-robot run test cases

**Review by dumganhar — APPROVED** [review]

**Comment by minggo** [comment]
@cocos-robot run test cases

**Pull request merged**